### PR TITLE
Add SSH Forwarding

### PR DESCRIPTION
**Pull request by goruha**
## What
* Added ssh listener

## Why
* Allow to ssh on eb servers

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -516,6 +516,29 @@ resource "aws_elastic_beanstalk_environment" "default" {
...
+    name      = "ListenerProtocol"
+    value     = "TCP"
+  }
+  setting {
+    namespace = "aws:elb:listener:22"
+    name      = "InstancePort"
+    value     = "22"
+  }
+  setting {
+    namespace = "aws:elb:listener:22"
```
Comment (osterman) [inline main.tf]: Can the listener port be a variable that defaults to `22`?

File: main.tf [hunk 2]
```diff
@@ -516,6 +516,29 @@ resource "aws_elastic_beanstalk_environment" "default" {
...
+    name      = "ListenerEnabled"
+    value     = "${var.ssh_listener_enabled}"
+  }
+
+  setting {
+    namespace = "aws:elb:policies"
+    name      = "ConnectionSettingIdleTimeout"
+    value     = "${var.ssh_listener_enabled == "true" ? "3600" : "60"}"
```
Comment (osterman) [inline main.tf]: can the policy be specific to the `listener`? e.g. `listener:22`

Comment (osterman) [reply main.tf]: Actually, we need the one for `v2` ELBs
image

Comment (osterman) [reply main.tf]: http://docs.aws.amazon.com/elasticbeanstalk/latest/dg/command-options-general.html

Comment (goruha) [reply main.tf]: For ssh we have to use classic elb. Because application elb does not allow to listen anything except http\s

Comment (goruha) [reply main.tf]: Also as you see this is param of loadbalancer not listner

File: main.tf [hunk 3]
```diff
@@ -516,6 +516,29 @@ resource "aws_elastic_beanstalk_environment" "default" {
...
+
+  setting {
+    namespace = "aws:elb:listener:${var.ssh_listener_port}"
+    name      = "ListenerProtocol"
+    value     = "TCP"
+  }
+  setting {
+    namespace = "aws:elb:listener:${var.ssh_listener_port}"
+    name      = "InstancePort"
+    value     = "22"
```
Comment (aknysh) [inline main.tf]: should "22" be replaced with ${var.ssh_listener_port} as well?

Comment (goruha) [reply main.tf]: no. Because on instance we listen port 22 always.

Comment (goruha) [reply main.tf]: ssh_listener_port -> forward -> 22

**Pull request merged**